File: WorldFeed/Server/src/WorldFeed.Common/Public/Models/Interfaces/DTO/IMessageDataDTO.cs
namespace WorldFeed.Common.Public.Models.Interfaces.DTO
{
    using WorldFeed.Common.Public.Models.Entities;

    public interface IMessageDataDTO
    {
        string Text { get; set; }

        IMessageEntities Entities { get; set; }

        IQuickReplyDTO QuickReply { get; set; }

        IQuickReplyResponse QuickReplyResponse { get; set; }

        IAttachmentDTO Attachment { get; set; }
    }
}

File: WorldFeed/Client/src/app/core/Public/Models/Interfaces/DTO/IMessageDataDTO.ts
import {IQuickReplyResponse} from "../IQuickReplyResponse";
import {IAttachmentDTO} from "./IAttachmentDTO";
import {IMessageEntities} from "../../Entities/IMessageEntities";
import {IQuickReplyDTO} from "./IQuickReplyDTO";

export interface IMessageDataDTO {
  text: string;
  entities: IMessageEntities;
  quickReply: IQuickReplyDTO;
  quickReplyResponse: IQuickReplyResponse;
  attachment: IAttachmentDTO;
}
